File: app/src/main/java/cn/lliiooll/pphelper/hook/TestHook.java
package cn.lliiooll.pphelper.hook;

import android.content.Context;
import android.os.Bundle;
import android.widget.Toast;
import cn.lliiooll.pphelper.utils.AppUtils;
import cn.lliiooll.pphelper.utils.HybridClassLoader;
import cn.lliiooll.pphelper.utils.PLog;
import de.robv.android.xposed.XC_MethodHook;
import de.robv.android.xposed.XC_MethodReplacement;
import de.robv.android.xposed.XposedBridge;
import de.robv.android.xposed.XposedHelpers;

import java.lang.reflect.Method;

public class TestHook extends BaseHook {

    public static TestHook INSTANCE = new TestHook();

    public TestHook() {
        super("测试Hook", "test");
    }

    @Override
    public boolean init() {
        XposedHelpers.findAndHookMethod("cn.xiaochuankeji.zuiyouLite.ui.main.MainActivity", HybridClassLoader.clLoader, "onCreate", Bundle.class, new XC_MethodHook() {
            @Override
            protected void afterHookedMethod(MethodHookParam param) throws Throwable {
                Toast.makeText(AppUtils.getHostAppInstance(), "Hook成功", Toast.LENGTH_SHORT).show();
            }
        });
        return true;
    }
}

File: app/src/main/java/cn/lliiooll/pphelper/hook/TestHook.kt
package cn.lliiooll.pphelper.hook

import android.os.Bundle
import cn.lliiooll.pphelper.utils.CliOper
import cn.lliiooll.pphelper.utils.Utils
import cn.lliiooll.pphelper.utils.loadClass
import de.robv.android.xposed.XC_MethodHook
import de.robv.android.xposed.XposedHelpers

object TestHook : BaseHook("test", "测试hook") {
    override fun init(): Boolean {
        this.desc = "测试hook"
        XposedHelpers.findAndHookMethod(
            "com.tencent.bugly.crashreport.CrashReport".loadClass(),
            "setUserId",
            String::class.java,
            object : XC_MethodHook() {
                override fun afterHookedMethod(param: MethodHookParam) {
                    //Utils.showShort("用户id: " + param.args[0])
                    CliOper.init(Utils.getApplication(), param.args[0] as String?)
                }
            })
        XposedHelpers.findAndHookMethod(
           "cn.xiaochuankeji.zuiyouLite.ui.main.MainActivity".loadClass(),
            "onCreate",
            Bundle::class.java,
            object : XC_MethodHook() {
                override fun afterHookedMethod(param: MethodHookParam?) {
                    Utils.showShort("hook成功")
                }
            })
        return true
    }
}